process.go

```diff
@@ -0,0 +1,154 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"io"
+
+	"github.com/urfave/cli"
+)
+
+// == Direct input processing ========================================
+
+func processInputFiles(c *cli.Context, g *Gomplate) error {
+	inputs, err := readInputs(c.String("in"), c.StringSlice("file"))
+	if err != nil {
+		return err
+	}
+
+	outputs := c.StringSlice("out")
+	if len(outputs) == 0 {
+		outputs = []string{"-"}
+	}
+
+	for n, input := range inputs {
+		if err := renderTemplate(g, input, outputs[n]); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// == Recursive input dir processing ======================================
+
+func processInputDir(c *cli.Context, g *Gomplate) error {
+	inputDir := c.String("input-dir")
+	outDir := c.String("output-dir")
+	if _, err := assertDir(outDir); err != nil {
```

Comment:
@hairyhenderson here is the assertion that the `outDir` must exist. I'm not sure if should we require this, but I think its common practice to require the top-level directory to exists. Otherwise typos in the usage can easily slip through leading to unexpected results.